sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MathFunctionsSuite.scala

```diff
@@ -252,4 +254,33 @@ class MathFunctionsSuite extends SparkFunSuite with ExpressionEvalHelper {
       null,
       create_row(null))
   }
+
+  test("round test") {
+    val domain = -16 to 16
+    val doublePi = math.Pi
+    val stringPi = "3.141592653589793"
+    val intPi = 314159265
+    val bdPi = BigDecimal(31415926535897932L, 10)
+
+    domain.foreach { scale =>
+      checkEvaluation(Round(doublePi, scale),
+        BigDecimal.valueOf(doublePi).setScale(scale, RoundingMode.HALF_UP).toDouble, EmptyRow)
```

Comment:
I think we are not comparing with itself, rather, checking the `Round` to see if it is branching to the right logic and run as expected.